### PR TITLE
feat(Subscription): Select by Subscriber in Subscriptions Editor

### LINKED ISSUE #592: Select by Subscriber in Subscriptions Editor
Here's a use case for a different interface than that provided in #549.

I'm not sure whether it should be in scope for that or a separate piece of work or not even be a requirement.

Here's what I was trying to do:

1. I had a sample file and I wanted to see the GOOSE subscriptions for a specific device.
2. I know the device name which has the subscriptions but I don't know the GOOSE publisher.

This means that with the information I have to hand I cannot locate the subscriptions.

Is this a reasonable use case? I think so. Let's imagine I am replacing a device and want to know "Everything it subscribes to".

Interested in the thoughts of others.

### PR DESCRIPTION
**Pull request by Flurb**
Closes #592

### CODE REVIEW

File: src/translations/en.ts [hunk 1]
```diff
@@ -309,15 +309,24 @@ export const en = {
...
+    },
+    subscriber: {
       subscribed: 'Subscribed',
       availableToSubscribe: 'Available to subscribe',
       partiallySubscribed: 'Partially subscribed',
       noGooseMessageSelected: 'No GOOSE message selected',
+      noIedSelected: 'No IED selected',
     },
+    view: {
+      publisherView: "GOOSE Publisher view",
```
Comment (JakobVogelsang) [inline src/translations/en.ts:327]: The names are not descriptive enough if you ask me. I would rather suggest something like:
`publisherView:  Show subscriber IED(s) per selected GOOSE`
`subscriberView:  Show subscribed GOOSE publisher for selected IED`

File: src/translations/de.ts [hunk 2]
```diff
@@ -313,14 +313,23 @@ export const de: Translations = {
...
+    },
+    subscriber: {
       subscribed: 'Verbunden',
       availableToSubscribe: 'Kann verbunden werden',
       partiallySubscribed: 'Teilweise verbunden',
       noGooseMessageSelected: 'Keine GOOSE ausgewählt',
+      noIedSelected: 'Keine IED ausgewählt',
     },
+    view: {
+      publisherView: "???",
```
Comment (JakobVogelsang) [inline src/translations/de.ts:330]: publisherView:  Zeigt verbundene IED(s) der ausgewählten GOOSE
subscriberView: Zeigt verbundene GOOSE(s) des ausgewählten IED

File: src/translations/en.ts [hunk 3]
```diff
@@ -309,15 +309,24 @@ export const en = {
...
     publisherGoose: {
-      title: 'GOOSE publisher',
+      title: 'GOOSE Publisher',
+      subscriberTitle: 'Subscriber of {{ selected }}'
     },
-    subscriberIed: {
-      title: 'Subscriber of {{ selected }}',
+    subscriberGoose: {
+      title: 'GOOSE Subscriber',
+      publisherTitle: 'Publisher {{ selected }}'
```
Comment (JakobVogelsang) [inline src/translations/en.ts:317]: GOOSE(s) subcriber by {{selected}}

File: src/translations/de.ts [hunk 4]
```diff
@@ -313,14 +313,23 @@ export const de: Translations = {
     none: 'Keine Verbindung vorhanden',
     publisherGoose: {
       title: 'GOOSE-Publizierer',
+      subscriberTitle: 'Verbunden mit {{ selected }}'
     },
-    subscriberIed: {
-      title: 'Verbunden mit {{ selected }}',
+    subscriberGoose: {
+      title: 'IED-Publizierer',
+      publisherTitle: '??? {{ selected }}'
```
Comment (JakobVogelsang) [inline src/translations/de.ts:320]: GOOSE(s) verbunden mit {{selected}}

**Comment by Flurb** [comment]
@JakobVogelsang is the review done besides the translations?

File: test/unit/editors/subscription/subscriber-ied-list.test.ts [hunk 5]
```diff
@@ -1,20 +1,20 @@
 import { html, fixture, expect } from '@open-wc/testing';
```
Comment (dlabordus) [inline test/unit/editors/subscription/subscriber-ied-list.test.ts:1]: Shouldn't this test be renamed to 'subscriber-list.test.ts' and also re-generated the snapshot file.

File: test/unit/editors/subscription/__snapshots__/subscriber-ied-list.test.snap.js [hunk 6]
```diff
@@ -4,7 +4,7 @@ export const snapshots = {};
 snapshots["subscriber-ied-list initially looks like the latest snapshot"] =
```
Comment (dlabordus) [inline test/unit/editors/subscription/__snapshots__/subscriber-ied-list.test.snap.js:4]: Test is only creating 1 snapshot, there are 2 in here.

File: test/unit/editors/subscription/goose-subscriber-list.test.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    element = await fixture(html`<goose-subscriber-list
+      .doc=${validSCL}
+    ></goose-subscriber-list>`);
+  });
+
+  it('looks like the latest snapshot', async () => {
+    await expect(element).shadowDom.to.equalSnapshot();
+  });
+
+  it('looks like the latest snapshot without a doc loaded', async () => {
```
Comment (dlabordus) [inline test/unit/editors/subscription/goose-subscriber-list.test.ts:24]: The before each is now also execute, but nothing is used from it. 
Maybe but two extra describe('') to the test, one called 'without document' and the other 'with document'.

File: src/translations/en.ts [hunk 8]
```diff
@@ -309,15 +309,24 @@ export const en = {
...
     publisherGoose: {
-      title: 'GOOSE publisher',
+      title: 'GOOSE Publisher',
+      subscriberTitle: 'Subscriber of {{ selected }}'
     },
-    subscriberIed: {
-      title: 'Subscriber of {{ selected }}',
+    subscriberGoose: {
+      title: 'GOOSE Subscriber',
+      publisherTitle: 'GOOSE(s) subscriber by {{selected}}'
```
Comment (JakobVogelsang) [inline src/translations/en.ts:317]:
```suggestion
      publisherTitle: 'GOOSE(s) subscribed by {{selected}}'
```

File: src/editors/subscription/elements/goose-publisher-list.ts [hunk 9]
```diff
@@ -11,10 +11,10 @@ import { translate } from 'lit-translate';
 import '@material/mwc-icon';
 import '@material/mwc-list/mwc-list-item';
 
-import '../../filtered-list.js';
-import { compareNames, getNameAttribute } from '../../foundation.js';
-import { newGOOSESelectEvent, styles } from './foundation.js';
-import { gooseIcon } from '../../icons/icons.js';
+import '../../../filtered-list.js';
```
Comment (dlabordus) [inline src/editors/subscription/elements/goose-publisher-list.ts:14]: Why put these files in the directory elements and subscriber-list.ts is not there?

The test files are also not in there.

**Review by dlabordus — DISMISSED** [review]

File: test/unit/editors/subscription/subscriber-ied-list.test.ts (hunk 5 above)
Comment (Flurb) [reply test/unit/editors/subscription/subscriber-ied-list.test.ts:1]: Correct!

File: test/unit/editors/subscription/__snapshots__/subscriber-ied-list.test.snap.js (hunk 6 above)
Comment (Flurb) [reply test/unit/editors/subscription/__snapshots__/subscriber-ied-list.test.snap.js:4]: A leftover, now it's removed

File: test/integration/editors/subscription/Subscription.test.ts [hunk 10]
```diff
@@ -24,84 +24,219 @@ describe('Subscription Plugin', () => {
...
-    });
 
-    describe('and you unsubscribe a subscribed IED', () => {
-      it('it looks like the latest snapshot', async () => {
-        const ied = Array.from(
-          element.shadowRoot
-            ?.querySelector('subscriber-ied-list-goose')
-            ?.shadowRoot?.querySelectorAll('mwc-list-item') ?? []
-        ).filter(item => !item.noninteractive)[1];
+      describe('and you subscribe a non-subscribed IED', () => {
```
Comment (JakobVogelsang) [inline test/integration/editors/subscription/Subscription.test.ts:66]: the description up until now reads like this: 
`Subscription Plugin -> in GOOSE publisher view -> when selecting a GOOSE message -> and you subscribe a non-subscribed IED`. The last description would read better like so `and subscribing a unsubscribed IED`.

This is true also for the next two descriptions :)

File: test/unit/editors/subscription/goose-subscriber-list.test.ts (hunk 7 above)
Comment (Flurb) [reply test/unit/editors/subscription/goose-subscriber-list.test.ts:24]: Done

File: test/integration/editors/subscription/Subscription.test.ts (hunk 10 above)
Comment (Flurb) [reply test/integration/editors/subscription/Subscription.test.ts:66]: Done!

File: src/editors/subscription/elements/goose-publisher-list.ts (hunk 9 above)
Comment (Flurb) [reply src/editors/subscription/elements/goose-publisher-list.ts]: Done!

File: test/integration/editors/subscription/Subscription.test.ts [hunk 11]
```diff
@@ -24,84 +24,219 @@ describe('Subscription Plugin', () => {
...
+          ).filter(item => !item.noninteractive)[2];
+
+          (<HTMLElement>ied).click();
+          await element.updateComplete;
+
+          // Re select the GOOSE
+          (<HTMLElement>goose).click();
+          await element.updateComplete;
+
+          await expect(
```
Comment (JakobVogelsang) [inline test/integration/editors/subscription/Subscription.test.ts:81]: The snapshot test is nice here. I would however prefer to have a check in the doc as well. Meaning an additional ```ts
it('add required connections (ExtRef) to the subscriber IED',()=>{
...     
expect(doc.querySelector...).to.equal(...)
}) 

```

**Review by JakobVogelsang — CHANGES_REQUESTED** [review]
Some additional tests. Other than that 👍

File: test/integration/editors/subscription/Subscription.test.ts (hunk 11 above)
Comment (Flurb) [reply test/integration/editors/subscription/Subscription.test.ts:83]: Done!

**Review by JakobVogelsang — APPROVED** [review]
LGTM

**Pull request merged**